src/main/java/org/dependencytrack/resources/v1/DependencyGraphResource.java

```diff
@@ -0,0 +1,232 @@
+import io.swagger.annotations.ApiResponses;
+import io.swagger.annotations.Authorization;
+import org.dependencytrack.auth.Permissions;
+import org.dependencytrack.model.Component;
+import org.dependencytrack.model.Project;
+import org.dependencytrack.model.RepositoryMetaComponent;
+import org.dependencytrack.model.RepositoryType;
+import org.dependencytrack.persistence.QueryManager;
+import org.dependencytrack.persistence.RepositoryQueryManager;
+import org.dependencytrack.resources.v1.vo.DependencyGraphResponse;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * JAX-RS resources for processing requests related to DependencyGraph.
+ */
+@Path("/v1/dependencyGraph")
+@Api(value = "dependencyGraph", authorizations = @Authorization(value = "X-Api-Key"))
+public class DependencyGraphResource extends AlpineResource {
+
+    @GET
+    @Path("/project/{uuid}/directDependencies")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(
+            value = "Returns a list of specific components and services from project UUID",
+            response = DependencyGraphResponse.class,
+            responseContainer = "List"
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 401, message = "Unauthorized"),
+            @ApiResponse(code = 403, message = "Access to a specified component is forbidden"),
+            @ApiResponse(code = 404, message = "Any component can be found"),
+    })
+    @PermissionRequired(Permissions.Constants.VIEW_PORTFOLIO)
+    public Response getComponentsAndServicesByProjectUuid(final @PathParam("uuid") String uuid) {
+        try (QueryManager qm = new QueryManager()) {
+            final Project project = qm.getObjectByUuid(Project.class, uuid);
+
+            if (project == null) {
+                return Response.status(Response.Status.NOT_FOUND).entity("The component could not be found.").build();
+            }
+
+            if (!qm.hasAccess(super.getPrincipal(), project)) {
+                return Response.status(Response.Status.FORBIDDEN).entity("Access to the specified project is forbidden").build();
+            }
+
+            final String directDependenciesJSON = project.getDirectDependencies();
+
+            if (directDependenciesJSON != null) {
+                final List<DependencyGraphResponse> response = getDependencyGraphFromDirectDependenciesJSON(qm, directDependenciesJSON);
+                return Response.ok(response).build();
+            } else {
+                return Response.ok(List.of()).build();
+            }
+        }
```

Comment:
@sonatype-lift ignore

The resource ProjectQueryManager is not released, but it doesn't have to.
The only critical resource of ProjectQueryManager is the PersistenceManager which is the same instance as QueryManager. So when QueryManager is close, all critical resources of ProjectQueryManager are closed.
The QueryManager is close at the end of the try scope because it's a try-with-resources.